db/index.js

```diff
@@ -2,65 +2,104 @@
 
 const { Pool } = require("pg");
 
-const pool = new Pool({
-  user: process.env.DB_USER,
-  host: process.env.DB_HOST,
-  database: process.env.DB_DATABASE,
-  password: process.env.DB_PASSWORD,
-  port: process.env.DB_PORT,
-});
+let database;
 
-/**
- * createTable()
- * Create the "barcodes" table if it's not already in the database.
- */
-async function createTable() {
-  const query = `
-    CREATE TABLE barcodes (
-      barcode VARCHAR(15) PRIMARY KEY NOT NULL,
-      used BOOLEAN DEFAULT false,
-      created_at timestamp DEFAULT CURRENT_TIMESTAMP,
-      updated_at timestamp DEFAULT CURRENT_TIMESTAMP
-    );`;
-  try {
-    const res = await pool.query(query);
-    if (res.command === "CREATE") {
-      console.log("database table 'barcodes' created");
-    }
-  } catch (error) {
-    if (error.message === 'relation "barcodes" already exists') {
-      console.log("database table barcodes already exists, continuing");
+class BarcodesDb {
+  constructor(args) {
+    this.pool = new Pool({
+      user: args["user"],
+      host: args["host"],
+      database: args["database"],
+      password: args["password"],
+      port: args["port"],
+    });
+  }
```

Comment:
What's the difference between BarcodesDb here and BarcodeDb in app.js?

Reply:
This is the class. When this file is imported and invoked, an instance will be created and returned. The name in app.js is misleading and I can update that, but that `BarcodeDb` is the wrapper function that returns the instance of this class. I did it this way because I only wanted to have one instance of the class that connects to the database for every startup of the lambda. That way, multiple requests will use the same instance instead of creating a new instance for every request.